src/TypeSpec.Extension/Emitter.Csharp/src/lib/client-model-builder.ts

```diff
@@ -157,8 +157,16 @@ export function createModelForService(
     const clients: InputClient[] = [];
     const dpgClients = listClients(sdkContext);
     for (const client of dpgClients) {
-        clients.push(emitClient(client));
-        addChildClients(sdkContext.emitContext, client, clients);
+        const emittedClient = emitClient(client, sdkContext.arm);
```

Comment:
I assume it should not be in @typespec/http-client-csharp? How you are going to keep it after https://github.com/Azure/autorest.csharp/pull/4774?

Reply:
Please check the latest commit.

Reply:
Reverted the change in emitter, move the logic to generator instead.